src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1279,14 +1278,17 @@ private Deadline NewAckDeadline()
             return Deadline.Now + Settings.SysMsgAckTimeout;
         }
 
-        private void PublishAndThrow(Exception reason, LogLevel level)
+        private void PublishAndThrow(Exception reason, LogLevel level, bool needToThrow = true)
         {
             reason.Match()
                 .With<EndpointDisassociatedException>(endpoint => PublishDisassociated())
                 .With<ShutDownAssociation>(shutdown => {}) // don't log an error for planned shutdowns
                 .Default(msg => PublishError(reason, level));
 
-            throw reason;
+            if (needToThrow)
+            {
+                throw reason;
```

Comment:
I agree, rethrowing this exception in the `SupervisionStrategy` is not the right call for this issue. Good choice.